go/service/user.go

```diff
@@ -594,3 +612,27 @@ func (h *UserHandler) LoadHasRandomPw(ctx context.Context, arg keybase1.LoadHasR
 
 	return ret.RandomPW, err
 }
+
+func (h *UserHandler) CanLogout(ctx context.Context, sessionID int) (res keybase1.CanLogoutRes, err error) {
+	hasRandomPW, err := h.LoadHasRandomPw(ctx, keybase1.LoadHasRandomPwArg{
+		SessionID:   sessionID,
+		ForceRepoll: false,
+	})
+
+	if err != nil {
+		return keybase1.CanLogoutRes{
+			CanLogout: false,
+			Reason:    fmt.Sprintf("Cannot check user state: %s", err.Error()),
```

Comment:
i think this would be better with a generic error message that is user friendly with internal logging of the raw error

Reply:
Thanks, you are right. I will change that in the frontend ticket that I have in TODO to return some kind of special variable here that frontend can react differently to, instead of stuffing everything together in the reason string.